### PR TITLE
feat: Autocompletion hints in sql editor

### LINKED ISSUE #17441: [Feature]: Autocomplete Feature for SQL Commands, Tables and Columns
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Users should be able to leverage autocomplete capabilities when writing SQL commands, Tables and Column names within the query editors.

### Why should this be worked on?

This would provide a better Query writing experience for our users.

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

This PR introduces autocompletion hints in the SQL editor

Fixes #17441 

Media
Screenshot 2023-05-07 at 14 31 11

Screenshot 2023-05-07 at 14 31 48

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

- Manual
- Jest
- Cypress

### Test Plan
https://github.com/appsmithorg/TestSmith/issues/2381

### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/22827#issuecomment-1536164809

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by ramsaptami** [comment]
- [ ] Create database, alter database don't show up on autocomplete

**Comment by ramsaptami** [comment]
lgtm
cc: @ohansFavour

**Review by arunvjn — APPROVED** [review]

**Review by arunvjn — APPROVED** [review]

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Autocomplete/Autocomplete_sql_spec.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const queryLocators = require("../../../../locators/QueryEditor.json");
+const datasource = require("../../../../locators/DatasourcesEditor.json");
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+
+const locator = ObjectsRegistry.CommonLocators;
+let datasourceName;
+
+describe("SQL Autocompletion", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Autocomplete/Autocomplete_sql_spec.js:8]: Why are tests using js helper?

Comment (ohansFavour) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Autocomplete/Autocomplete_sql_spec.js:8]: I will make this change in another PR.

**Pull request merged**